### PR TITLE
make peft installs a extra_dep

### PR DESCRIPTION
**Pull request by vchiley**
If I run anything I get this warning:
```
===================================BUG REPORT===================================
Welcome to bitsandbytes. For bug reports, please run

python -m bitsandbytes

 and submit this information together with your error trace to: https://github.com/TimDettmers/bitsandbytes/issues
================================================================================
```

this gates the parameter efficient fine tuning installs so that the above only happens if a user wants it to.

### CODE REVIEW

File: llmfoundry/models/hf/hf_causal_lm.py [hunk 1]
```diff
@@ -24,6 +23,12 @@
 from llmfoundry.models.hf.model_wrapper import HuggingFaceModelWithZLoss
 from llmfoundry.models.utils import init_empty_weights
 
+try:
+    from peft.peft_model import PeftModel
+    model_types = (PeftModel, transformers.PreTrainedModel)
+else:
```
Comment (samhavens) [inline llmfoundry/models/hf/hf_causal_lm.py:29]:
```suggestion
except ImportError:
```

**Review by samhavens — COMMENTED** [review]
I think `else -> except`, otherwise LGTM

File: llmfoundry/models/hf/hf_causal_lm.py (hunk 1 above)
Comment (vchiley) [reply llmfoundry/models/hf/hf_causal_lm.py]: wanted to push before a mtg.
that was hilarious 🤣

**Review by samhavens — APPROVED** [review]
LGTM

**Review by samhavens — APPROVED** [review]
LGTM2: Return of the LGTM

**Pull request merged**